src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class SalaryInfo {
+    private final Pattern datePattern = Pattern.compile("\\d{2}\\.\\d{2}\\.\\d{4}");
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
+        LocalDate fromDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate toDate = LocalDate.parse(dateTo, formatter);
+        StringBuilder report = new StringBuilder("Report for period ")
+                .append(dateFrom).append(" - ").append(dateTo).append(System.lineSeparator());
+
+        for (String name : names) {
+            int totalSalary = 0;
+            for (String record : data) {
+                if (record.contains(name)) {
+                    LocalDate date = extractDate(record);
```

Comment:
The `extractDate` method returns `null` if the date is not found or if there is a parsing error. Ensure that the `record` string always contains a valid date to avoid `NullPointerException` when calling methods on the `date` object.